src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -114,14 +122,38 @@ static bool LocationInvalid(int location, SyntaxNode node)
         }
 
         // Correct selection to include the current node if the selection ends immediately after a closing tag.
-        if (endOwner is MarkupTextLiteralSyntax
-            && endOwner.ContainsOnlyWhitespace()
+        if (endOwner is MarkupTextLiteralSyntax markupTextLiteral
+            && SelectionShouldBePrevious(markupTextLiteral, context.EndAbsoluteIndex)
             && endOwner.TryGetPreviousSibling(out var previousSibling))
         {
             endOwner = previousSibling;
         }
 
-        return endOwner.FirstAncestorOrSelf<SyntaxNode>(IsBlockNode);
+        return GetBlockOrTextNode(endOwner);
+
+        static bool SelectionShouldBePrevious(MarkupTextLiteralSyntax markupTextLiteral, int absoluteIndex)
+            => markupTextLiteral.Span.Start == absoluteIndex
+                || markupTextLiteral.ContainsOnlyWhitespace();
```

Comment:
Might make it clearer if this just too `context`, and had a comment, as in isolation (ie, outside of the context of this PR) I'm not sure the name makes sense when looking at the logic. There is nothing obviously "previous" about the code.